deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/DeployTagWorker.java

```diff
@@ -122,14 +124,17 @@ private void processBatch() throws Exception {
                 LOG.info("process job: {}", job);
                 String lockName = String.format("DeployTagWorker-%s", job.getConstraint_id());
                 Connection connection = utilDAO.getLock(lockName);
+                DeployConstraintBean latestJob = deployConstraintDAO.getById(job.getConstraint_id());
```

Comment:
do we need this? the job and job_id should be 1-1 and not mutable.

Reply:
@sulixu mentioned that there is a good chance that `job` may get mutated by a worker by the time it enters this state. The idea is to acquire lock first, then query the job to make sure mutations on it don't happen.``

Reply:
nvm -- I agree w/ your comment. I'll revert the latest job code